lib/matplotlib/colorizer.py

```diff
@@ -476,31 +472,54 @@ def _format_cursor_data_override(self, data):
 
         # Note if cm.ScalarMappable is depreciated, this functionality should be
         # implemented as format_cursor_data() on ColorizingArtist.
-        n = self.cmap.N
-        if np.ma.getmask(data):
+        if np.ma.getmask(data) or data is None:
             return "[]"
-        normed = self.norm(data)
+        if len(data.dtype.descr) > 1:
+            # We have multivariate data encoded as a data type with multiple fields
+            # NOTE: If any of the fields are masked, "[]" would be returned via
+            # the if statement above.
+            s_sig_digits_list = []
+            if isinstance(self.cmap, colors.BivarColormap):
+                n_s = (self.cmap.N, self.cmap.M)
+            else:
+                n_s = [part.N for part in self.cmap]
+            os = [f"{d:-#.{self._sig_digits_from_norm(no, d, n)}g}"
+                  for no, d, n in zip(self.norm.norms, data, n_s)]
+            return f"[{', '.join(os)}]"
+
+        # scalar data
+        n = self.cmap.N
+        g_sig_digits = self._sig_digits_from_norm(self.norm, data, n)
+        return f"[{data:-#.{g_sig_digits}g}]"
```

Comment:
Since scalar data gets the braces too, I think you could set `n_s = [self.cmap.N]; data = [data]` and use the same `os` from the multivariate case for both.